lib/mix/tasks/triplex.migrate.ex

```diff
@@ -1,116 +1,45 @@
-
-  Runs all pending migrations by default. To migrate up
-  to a version number, supply `--to version_number`.
-  To migrate up a specific number of times, use `--step n`.
-
-  If the repository has not been started yet, one will be
-  started outside our application supervision tree and shutdown
-  afterwards.
-
-  ## Examples
-
-      mix ecto.migrate
-      mix ecto.migrate -r Custom.Repo
-
-      mix ecto.migrate -n 3
-      mix ecto.migrate --step 3
-
-      mix ecto.migrate -v 20080906120000
-      mix ecto.migrate --to 20080906120000
-
-  ## Command line options
+  It basically runs the task `ecto.migrate --prefix tenant` for all the
+  tenants existent on the db.
 
-    * `-r`, `--repo` - the repo to migrate
-    * `--all` - run all pending migrations
-    * `--step` / `-n` - run n number of pending migrations
-    * `--to` / `-v` - run all migrations up to and including version
-    * `--quiet` - do not log migration commands
-    * `--pool-size` - the pool size if the repository is started only for the
-    task (defaults to 1)
-
-  ## PS
-
-  All of this code is copied from `mix ecto.migrate` task, if something does
-  not work, please compare them and try to stay as close to it as possible.
+  All the options given to this task will be passed to the ecto.migrate
+  task, so for more information about the it check `mix help ecto.migrate`.
   """
 
-  @doc false
-  def run(args, migrator \\ &Migrator.run/4, testing? \\ false) do
+  @impl true
+  def run(args) do
     repos = parse_repo(args)
-
-    {opts, _, _} = OptionParser.parse args,
-      switches: [all: :boolean, step: :integer, to: :integer, quiet: :boolean,
-                 pool_size: :integer],
-      aliases: [n: :step, v: :to]
-
-    opts =
-      if opts[:to] || opts[:step] || opts[:all],
-        do: opts,
-        else: Keyword.put(opts, :all, true)
-
-    opts =
-      if opts[:quiet],
-        do: Keyword.put(opts, :log, false),
-        else: opts
-
-    Enum.each repos, fn repo ->
-      ensure_repo(repo, args)
-      ensure_tenant_migrations_path(repo)
-      {:ok, pid, apps} = ensure_started(repo, opts)
-
-      # If the pool is Ecto.Adapters.SQL.Sandbox,
-      # let's make sure we get a connection outside of a sandbox.
-      if sandbox?(repo) and !testing? do
-        Sandbox.checkin(repo)
-        Sandbox.checkout(repo, sandbox: false, ownership_timeout: :infinity)
-      end
-
-      Code.compiler_options(ignore_module_conflict: true)
-      migrated = Enum.reduce Triplex.all(repo), [], fn(tenant, acc) ->
-        migrate_tenant(opts, migrator, repo, tenant, acc)
+    for repo <- repos do
+      {opts, _} = OptionParser.parse!(args, strict: [pool_size: :integer])
+      ^repo = ensure_repo(repo, args)
+      {:ok, pid, _apps} = ensure_started(repo, opts)
+
+      for prefix <- Triplex.all(repo) do
+        new_args = delete_unwanted_args(args, [])
+        Mix.Task.run("ecto.migrate", ["--repo", inspect(repo), "--prefix", prefix | new_args])
```

Comment:
I totally forgot this don't work. :(

This way it does not run the migrations inside the `tenant_migrations` folder, and yes on `migrations` folder.

Maybe we should change that too, for the new version of triplex. Instead of having a `tenant_migrations` folder, we could put all migrations inside `migrations` and if someone would not like to filter out migrations on public or tenant schemas one could do somthing like:

```
def change do
  if Triplex.reserved_tenant?(prefix()) do
    # run your public schema migration
  else
    # run your tenant schema migration
  end
end
```

Reply:
@mgiacomini what do you think? We would then totally break the current working version. The dump change would break it anyways, but the solution for that would be just to run a `triplex.dump`, this one will take a lot more trouble for users to port to the new version, since they would have to add an `if` inside all of their tenant migrations.

Reply:
The other solution would be rolling back to the old way of running migrations, but that's a trouble for us then, since any change on the migration stuff on ecto's side, we would have to change ours too.

Reply:
So, looks like the change to use only one migrations folder is not possible at all, and we will have to make it work with `tenant_migrations`. I'll delete my commits til now since they are all in favor of the idea of using `ecto.migrate --prefix`, and that will not be possible at all.

Reply:
@kelvinst was the migration fixed?
I can't run `mix triplex.migrate`

Reply:
> @kelvinst was the migration fixed?
> I can't run mix triplex.migrate

Where are you running them? Is it your project? Are you pointing the dep to this PR branch?

> How does one pass a default tenant name to migrations to ensure migrations are run for a potentially pre-existing tenant(s)?

That's pretty much what `mix triplex.migrate` does. It runs all the migrations for all the pre-existing tenants, not for new ones. Migration for the new ones a ran when you call `Triplex.create`